From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 30 Nov 2022 14:39:27 +0100
Subject: [PATCH] Hide flames on entities with fire resistance

License: MIT (https://opensource.org/licenses/MIT)
Gale - https://galemc.org

This patch is based on the following patch:
"Don't send fire packets if player has Fire Resistance"
By: Cryptite <cryptite@gmail.com>
As part of: Slice (https://github.com/Cryptite/Slice)
Licensed under: MIT (https://opensource.org/licenses/MIT)

diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 9f674f579a3a0f6fe9eb71fa2cace7f0a93d5702..ec303fb52aee7a0bc1df42a628fac24276d059de 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -884,7 +884,13 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
 
         this.checkBelowWorld();
         if (!this.level().isClientSide) {
+            // Gale start - Slice - hide flames on entities with fire resistance
+            if (this instanceof net.minecraft.world.entity.LivingEntity livingEntity) {
+                this.setSharedFlagOnFire(this.remainingFireTicks > 0 && (!this.level.galeConfig().gameplayMechanics.hideFlamesOnEntitiesWithFireResistance || !livingEntity.hasEffect(net.minecraft.world.effect.MobEffects.FIRE_RESISTANCE)));
+            } else {
+            // Gale end - Slice - hide flames on entities with fire resistance
             this.setSharedFlagOnFire(this.remainingFireTicks > 0);
+            } // Gale - Slice - hide flames on entities with fire resistance
         }
 
         this.firstTick = false;
